src/transformers/modeling_utils.py

```diff
@@ -1239,6 +1243,24 @@ def _backward_compatibility_gradient_checkpointing(self):
             # Remove the attribute now that is has been consumed, so it's no saved in the config.
             delattr(self.config, "gradient_checkpointing")
 
+    def set_model_tags(self, tags: Union[List[str], str]) -> None:
+        r"""
+        Manually set the model tags with `tags`
+
+        Args:
+            tags (`Union[List[str], str]`):
+                The desired tags to inject in the model
+        """
+        if isinstance(tags, str):
+            tags = [tags]
+
+        if self._model_tags is None:
+            self._model_tags = []
+
+        for tag in tags:
+            if tag not in self._model_tags:
+                self._model_tags.append(tag)
```

Comment:
I don't think we want an append behaviour here - more of an overwrite. Otherwise, if I accidentally add a tag there's no way to ever remove it. I'd suggest a `set_model_tags` and `add_model_tags` method

Reply:
I'd also be in favor of switching to `add_model_tags` if we are appending tags to a list.
Don't know if adding support for removing tags if really necessary (but I don't have full context).

Reply:
Yeah makes sense, I changed that method to `add_tags` and added a new method `set_tags`